### PR TITLE
Add list unit files method

### PR DESCRIPTION
**Pull request by ranjib**
I need the ListUnitFiles method, along side several other systemd dbus calls for one of our project. Request for this method has already been mentioned here https://github.com/coreos/go-systemd/issues/76 . Im new to dbus, let me know if the dbus.Store call is correct, I have tested in on ubuntu 15.10 running systemd 225

### CODE REVIEW

**Comment by ranjib** [comment]
@jonboulle any thoughts on this one?

**Comment by ranjib** [comment]
@jonboulle  ping

File: dbus/methods.go [hunk 1]
```diff
@@ -250,6 +249,37 @@ func (c *Conn) ListUnits() ([]UnitStatus, error) {
...
+
+	resultInterface := make([]interface{}, len(result))
+	for i := range result {
+		resultInterface[i] = result[i]
+	}
+
+	files := make([]UnitFile, len(result))
+	fileInterface := make([]interface{}, len(files))
+
+	for i := range result {
```
Comment (jonboulle) [inline dbus/methods.go]: For strict consistency it'd be better to `range files` here

File: dbus/methods.go [hunk 2]
```diff
@@ -250,6 +249,37 @@ func (c *Conn) ListUnits() ([]UnitStatus, error) {
...
+		fileInterface[i] = &files[i]
+	}
+	err = dbus.Store(resultInterface, fileInterface...)
+	if err != nil {
+		return nil, err
+	}
+	return files, nil
+}
+
 type UnitStatus struct {
```
Comment (jonboulle) [inline dbus/methods.go]: This should be closer to the ListUnits() func (actually I'd prefer both the UnitStatus and UnitFile struct defs were _above_ the respective funcs, not below them)

File: dbus/methods.go [hunk 3]
```diff
@@ -250,6 +249,37 @@ func (c *Conn) ListUnits() ([]UnitStatus, error) {
 	return status, nil
 }
 
+type UnitFile struct {
+	Path string
+	Type string
+}
+
+// ListUnitFiles returns an array of with all available units in disk.
```
Comment (jonboulle) [inline dbus/methods.go]: array of all available units on disk

File: dbus/methods.go [hunk 4]
```diff
@@ -16,10 +16,9 @@ package dbus
 
 import (
 	"errors"
+	"github.com/godbus/dbus"
```
Comment (jonboulle) [inline dbus/methods.go]: Hm, please revert, goimports shouldn't do this

**Comment by jonboulle** [comment]
Sorry, lgtm overall, but a few nits

**Comment by jonboulle** [comment]
landed via #135

**Pull request closed without merging**